### PR TITLE
Add Doc Example tests for FileBasedSystemAccessControl

### PR DESCRIPTION
**Pull request by haldes**
@kokosing @findepi @dain 

Follow up from this discussion [3952]( https://github.com/trinodb/trino/issues/3952)

I have added the test cases for the example docs for FileBasedSystemAccessControl.
- Catalog Rules
- Schema Rules
- Table Rules

### CODE REVIEW

File: docs/src/main/sphinx/security/catalog_rules.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (kokosing) [inline docs/src/main/sphinx/security/catalog_rules.json:1]: @mosabua Can we use these files as fragments in docs? Otherwise it is easy to get out of sync.

Comment (mosabua) [reply docs/src/main/sphinx/security/catalog_rules.json:1]: You can import them or at least link to them.

Comment (mosabua) [reply docs/src/main/sphinx/security/catalog_rules.json:1]: See https://github.com/trinodb/trino/blob/master/docs/src/main/sphinx/admin/resource-groups.rst

```
.. literalinclude:: resource-groups-example.json
    :language: json
```

Comment (haldes) [reply docs/src/main/sphinx/security/catalog_rules.json:1]: Thanks @mosabua  @kokosing 
Will make the change and get back.

**Pull request closed without merging**